src/main/kotlin/org/rust/cargo/toolchain/Cargo.kt

```diff
@@ -109,6 +110,12 @@ class Cargo(
             .withParameters(commandLine.command)
             .withEnvironment(CargoConstants.RUSTC_ENV_VAR, rustExecutable.toString())
 
+        val proxyHelper = ProxyHelper(HttpConfigurable.getInstance())
+
+        if (proxyHelper.isHttpProxyEnabled()) {
+            cmdLine.withEnvironment("http_proxy", proxyHelper.getProxyUri().toString())
+        }
```

Comment:
Let's add a unit-test style test for this? It could work along this lines:

1. Set proxy settings
2. Create a `Cargo` and `CargoCommandLine` instances
3. Call `toGeneralCommandLine`
4. Observe that proxy env var is present.